eclair-core/src/main/scala/fr/acinq/eclair/Eclair.scala

```diff
@@ -70,13 +70,16 @@ class EclairImpl(appKit: Kit) extends Eclair {
     (appKit.switchboard ? Peer.Connect(NodeURI.parse(uri))).mapTo[String]
   }
 
-  override def open(nodeId: PublicKey, fundingSatoshis: Long, pushMsat: Option[Long], fundingFeerateSatByte: Option[Long], flags: Option[Int]): Future[String] = {
+  override def open(nodeId: PublicKey, fundingSatoshis: Long, pushMsat: Option[Long], fundingFeerateSatByte: Option[Long], flags: Option[Int], timeout_opt: Option[Timeout]): Future[String] = {
+    // we want the default timeout to expire *before* the default ask times out, otherwise user won't get a usable response
+    val openTimeout = timeout_opt.getOrElse(Timeout(30 seconds))
     (appKit.switchboard ? Peer.OpenChannel(
```

Comment:
Why not setting an explicit timeout here?

Reply:
Good catch, addressed with 59ede841.